libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1193,19 +1179,49 @@ std::string YulUtilFunctions::resizeArrayFunction(ArrayType const& _type)
 			bool isMappingBase = _type.baseType()->category() == Type::Category::Mapping;
 			templ("needsClearing", !isMappingBase);
 			if (!isMappingBase)
-			{
-				templ("convertToSize", arrayConvertLengthToSize(_type));
-				templ("dataPosition", arrayDataAreaFunction(_type));
-				templ("clearStorageRange", clearStorageRangeFunction(*_type.baseType()));
-				templ("packed", _type.baseType()->storageBytes() <= 16);
-				templ("itemsPerSlot", to_string(32 / _type.baseType()->storageBytes()));
-				templ("storageBytes", to_string(_type.baseType()->storageBytes()));
-				templ("partialClearStorageSlot", partialClearStorageSlotFunction());
-			}
+				templ("cleanUpArrayEnd", cleanUpStorageArrayEndFunction(_type));
 			return templ.render();
 	});
 }
 
+string YulUtilFunctions::cleanUpStorageArrayEndFunction(ArrayType const& _type)
+{
+	solAssert(_type.location() == DataLocation::Storage, "");
+	solAssert(_type.baseType()->category() != Type::Category::Mapping, "");
+	solAssert(!_type.isByteArray(), "");
+	solUnimplementedAssert(_type.baseType()->storageBytes() <= 32, "");
+
+	string functionName = "cleanup_array_end_" + _type.identifier();
```

Comment:
Nitpicking so that the generated helper names reflect the C++ helper names:
```suggestion
	string functionName = "cleanup_storage_array_end_" + _type.identifier();
```